python/ray/serve/tests/test_proxy.py

```diff
@@ -1,11 +1,681 @@
+import pickle
+import sys
+from unittest.mock import patch, MagicMock, ANY
+from typing import AsyncGenerator
+
+import grpc
 import pytest
 import ray
 from ray import serve
 from ray.actor import ActorHandle
 from ray.serve._private.constants import (
     DEFAULT_UVICORN_KEEP_ALIVE_TIMEOUT_S,
+    SERVE_MULTIPLEXED_MODEL_ID,
     SERVE_NAMESPACE,
 )
+from ray.serve._private.http_proxy import (
+    GenericProxy,
+    gRPCProxy,
+    HTTPProxy,
+    HEALTH_CHECK_SUCCESS_MESSAGE,
+)
+from ray.serve._private.proxy_request_response import (
+    ASGIProxyRequest,
+    gRPCProxyRequest,
+    ProxyResponse,
+)
+from ray.serve._private.common import EndpointTag, RequestProtocol
+from ray.serve._private.constants import RAY_SERVE_ENABLE_EXPERIMENTAL_STREAMING
+from ray.serve.generated import serve_pb2
+
+if sys.version_info >= (3, 8, 0):
+    from unittest.mock import AsyncMock
+else:
+    from asyncmock import AsyncMock
+
+
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="requires python3.8 or higher")
+@patch("ray.serve._private.http_proxy.ray.get_runtime_context", MagicMock())
+@patch("ray.serve._private.http_proxy.ray.get_actor", MagicMock())
```

Comment:
please rewrite these tests so that we don't need to use mocks

in their place, implement fake classes that can be dependency-injected

see `test_replica_scheduler.py` for an example of how to do this. it defines a `FakeReplicaWrapper` which is used as an implementation of the `ReplicaWrapper` interface (as defined in `router.py`)

Reply:
good point, let me try it out!